examples/plot_openfmri_ds001.py

```diff
@@ -0,0 +1,214 @@
+            durations = []
+            # ASSUMING WE NEED AT LEAST TWO EVENTS FOR ANY CONDITION
+            if os.stat(cond_file).st_size > 0:
+                cond_info = pd.read_csv(cond_file,
+                                        sep=OPENFMRI_SEP,
+                                        header=None)
+                onsets = cond_info[0].tolist()
+                durations = cond_info[1].tolist()
+            else:
+                print 'empty file found: ' + cond_file
+                onsets = [vol*TR - 0.1]
+                durations = [0.0]
+            allonsets.append(onsets)
+            alldurations.append(durations)
+            n_on = len(list(itertools.chain.from_iterable(allonsets)))
+            n_dur = len(list(itertools.chain.from_iterable(alldurations)))
+            assert(n_on == n_dur)
+
+        _run_events[int(run_id)-1] = {}
+        _run_events[int(run_id)-1]['conditions'] = names
+        _run_events[int(run_id)-1]['onsets'] = allonsets
+        _run_events[int(run_id)-1]['durations'] = alldurations
+        all_event_files.append(event_files)
+
+    _subject_data['TR'] = TR
+    _subject_data['func'] = runs
+    _subject_data['anat'] = os.path.join(dataset_dir,
+                                         'sub{0:03d}'.format(sub_id),
+                                         'anatomy', 'highres001.nii.gz')
+    _subject_data['func_events'] = _run_events
+    _subject_data['onset_files'] = all_event_files
+
+    return Bunch(**_subject_data)
+
+
+def _openfrmri_contrasts(ref_design_matrix, dataset_dir, model_id, task_id):
+    # Build basic contrast_matrix.
+    contrast_matrix = np.eye(ref_design_matrix.shape[1])
+    contrasts = dict([(column, contrast_matrix[i])
+                      for i, column in enumerate(ref_design_matrix.columns)])
+
+    # Load information from openfmri files
+    task = 'task{0:03d}'.format(task_id)
+    condkey_file = os.path.join(dataset_dir, 'models',
+                                'model{0:03d}'.format(model_id),
+                                'condition_key.txt')
+    cond_df = pd.read_csv(condkey_file,
+                          sep=OTHER_FILE_SEP,
+                          header=None)
+    info_conditions = cond_df[cond_df[0] == task]
+    conds_name = info_conditions[2].tolist()
+    contrasts_file = os.path.join(dataset_dir, 'models',
+                                  'model{0:03d}'.format(model_id),
+                                  'task_contrasts.txt')
+    contrasts_df = pd.read_csv(contrasts_file,
+                               sep=OTHER_FILE_SEP,
+                               header=None)
+    task_contrasts = contrasts_df[contrasts_df[0] == task]
+
+    # Complete contrast specification
+    for idx, crow in task_contrasts.iterrows():
+        contrasts[crow[1]] = np.zeros(ref_design_matrix.shape[1])
+        for widx, weight in enumerate(crow[2:]):
+            contrasts[crow[1]] += weight*contrasts[conds_name[widx]]
+
+    return contrasts
+
```

Comment:
These ancillary functions are pretty ong and messy. They should be put in a side module to clarify the example.